app/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -595,14 +599,24 @@ private VideoCapturer createVideoCapturer() {
     }
 
     private VideoCapturer createCameraCapturer(CameraEnumerator enumerator) {
+        Logging.d(TAG, "Preparing stream capture.");
+
+        if(mMediaSource.equals(MEDIA_SOURCE_STREAM)){
```

Comment:
Thanks for this suggestion. I will update this to do regex check against the URL for being a valid RTSP url. I will also make the EditText show a red border until when the data is a valid RTSP url.

Reply:
I have added regex based validation for the URL string. It will show a red coloured border and a small message saying "Invalid RTSP URL" just below the text field when the URL is not valid. Kindly review.